machine_learning_hep/validation/validation_candidates.py

```diff
@@ -0,0 +1,51 @@
+#############################################################################
+##  © Copyright CERN 2018. All rights not expressly granted are reserved.  ##
+##                 Author: Gian.Michele.Innocenti@cern.ch                  ##
+## This program is free software: you can redistribute it and/or modify it ##
+##  under the terms of the GNU General Public License as published by the  ##
+## Free Software Foundation, either version 3 of the License, or (at your  ##
+## option) any later version. This program is distributed in the hope that ##
+##  it will be useful, but WITHOUT ANY WARRANTY; without even the implied  ##
+##     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    ##
+##           See the GNU General Public License for more details.          ##
+##    You should have received a copy of the GNU General Public License    ##
+##   along with this program. if not, see <https://www.gnu.org/licenses/>. ##
+#############################################################################
+
+"""
+Script containing validation histograms on the candidate granularity
+"""
+# pylint: disable=too-many-lines
+# pylint: disable=import-error, no-name-in-module
+from utilities_plot import buildbinning
+from machine_learning_hep.validation.validation import ValidationCollection
+
+
+def fill_validation_candidates(df_reco, tag=""):
+    """
+    Create histograms for the validation on the event level as a function of the multiplicity
+    """
+    _ = len(df_reco)
+
+    # Binning definition
+    binning_nsigma = buildbinning(1, -1000, -998)
+    binning_nsigma += buildbinning(2000, -100, 100)
+    binning_pt = buildbinning(100, 0, 100)
+    binning_eta = buildbinning(100, -1, 1)
+    binning_phi = buildbinning(100, 0, 7)
+
+    # Make and fill histograms
+    val = ValidationCollection(df_reco, tag=tag)
+    for i in "TPC TOF".split():
```

Comment:
Why do you not use
`for i in ("TPC", "TOF"):`

(same for following lines)

Reply:
does it change?

Reply:
Well, what I was suggesting I think would be the more straightforward way. Yours works of course but is something which is done at runtime (not sure whether the interpreter is smart enough to realise that this is something which can be resolved before runtime?!)

Reply:
You're right, that's a useless overhead